clash-cores/src/Clash/Cores/Xilinx/Floating/Explicit.hs

```diff
@@ -311,6 +316,45 @@ fromS32 = withFrozenCallStack fromS32With
 -- | The default delay for conversion of @Signed 32@ to @Float@
 type FromS32DefDelay = 6
 
+-- | Compare two floating point numbers. Produces 'NaN' if any of the inputs
+-- is NaN.
+compareWith
+  :: forall d dom n
+   . ( KnownDomain dom
+     , KnownNat d
+     , HasCallStack
+     )
+  => Clock dom
+  -> Enable dom
+  -> DSignal dom n Float
+  -> DSignal dom n Float
+  -> DSignal dom (n + d) Ordering
+compareWith clk ena a b = delayI und ena clk (xilinxCompare <$> a <*> b)
+ where
+  und = withFrozenCallStack $ errorX "Initial values of compare undefined"
+{-# NOINLINE compareWith #-}
+{-# ANN compareWith (vhdlComparePrim 'compareWith 'compareTclTF "compare") #-}
+{-# ANN compareWith (veriComparePrim 'compareWith 'compareTclTF "compare") #-}
+
+-- | Compare two floating point numbers. Produces 'Nothing' if any of the inputs
+-- is NaN. Otherwise, it behaves like Haskell's 'P.compare'. Function has a
+-- default delay, unlike 'compareWith'.
+compare
+  :: forall dom n
+   . ( KnownDomain dom
+     , HasCallStack
+     )
+  => Clock dom
+  -> Enable dom
+  -> DSignal dom n Float
+  -> DSignal dom n Float
+  -> DSignal dom (n + CompareDefDelay) Ordering
+compare = compareWith
+{-# INLINE compare #-}
+
+-- | The default delay for comparing @Float@ with @Float@
```

Comment:
```suggestion
-- | The default delay for @Float@ comparison
```

You compare something _to_ something else, but I like this phrasing even better

Reply:
> You compare something _to_ something else

[The dictionary](https://www.oxfordlearnersdictionaries.com/definition/english/compare_1) disagrees with (to?) me, both are possible, and from the examples, I get the sense "with" is more neutral (the examples with "to" are all about things that differ significantly).